game/src/contrib/utils/components/ai/AITools.java

```diff
@@ -97,15 +97,20 @@ public static List<Tile> accessibleTilesInRange(final Point center, final float
     }
 
     /**
-     * @param center center point
-     * @param radius search radius
-     * @return random tile in given range
+     * Gets a random accessible tile coordinate within a specified range around a given center
+     * point. The range is determined by the provided radius.
+     *
+     * @param center The center point around which the tiles are considered.
+     * @param radius The radius within which the accessible tiles should be located.
+     * @return An Optional containing a random Coordinate object representing an accessible tile
+     *     within the range, or an empty Optional if no accessible tiles were found.
      */
-    public static Coordinate randomAccessibleTileCoordinateInRange(
+    public static Optional<Coordinate> randomAccessibleTileCoordinateInRange(
             final Point center, final float radius) {
         List<Tile> tiles = accessibleTilesInRange(center, radius);
+        if (tiles.isEmpty()) return Optional.empty();
         Coordinate newPosition = tiles.get(random.nextInt(tiles.size())).coordinate();
-        return newPosition;
+        return Optional.of(newPosition);
```

Comment:
Brauchen wir hier noch die Hilfsvariable `newPosition`? Im Prinzip könnte man das Zufallsexperiment doch gleich in das `Optional.of()` reinpacken, oder?

Reply:
@AMatutat Ich würde sogar noch einen Schritt weitergehen und tatsächlich `Optional<Tile>` zurückliefern und dem Aufrufer überlassen, sich die Coordinate oder den Point da rauszuprökeln. Also `public static Optional<Tile> randomAccessibleTileInRange(` ... Liest sich auch gleich viel besser :)

Reply:
@cagix #787